### PR TITLE
Reworking Azure Developer Hub for consistency with other languages (alternate)

### LINKED ISSUE #18891: Align Azure for .NET Developers to Azure for Python Developers
- [x] Consolidate existing hub (dotnet/azure) and landing page (dotnet/azure/sdk) into single landing page
- [x] Construct new TOC
- [x] Links for Messaging tile on index.yml
- [x] Links for Diagnostics and Monitoring tile on index.yml
- [x] Review authentication.md and re-write if needed.
    - Is it needed? We might just link [Microsoft.Azure.Identity docs](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme?view=azure-dotnet).
- [x] Evaluate need for *Explore specific Azure services* curated links pages [like this one](https://docs.microsoft.com/en-us/azure/developer/python/quickstarts-app-hosting).
    - My unsolicited opinion: Not necessary.
- Rewrite example tutorials from Python to .NET (may make use of includes to cover common information, like service principals)
    - [ ] database.md
    - [ ] resource-group.md
    - [ ] storage.md
    - [ ] storage-use.md
    - [ ] virtual-machines.md
    - [ ] web-app.md
- [ ] Redirects for pages formerly in azure/sdk
- [ ] Present to stakeholders for feedback

cc: @IEvangelist

### PR DESCRIPTION
**Pull request by CamSoper**
Contributes to #18891 .  Alternate version using hub layout.

# Preview

✔️ [New landing page](https://review.docs.microsoft.com/dotnet/azure/?branch=pr-en-us-19358)
✔️ [Old landing page for reference](https://docs.microsoft.com/dotnet/azure/)

### CODE REVIEW

File: docs/azure/index.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ### YamlMime:Hub 
 
 title: Azure for .NET developers 
-summary: Find the information you need to use .NET on Azure 
+summary: .NET 💜 Azure
```
Comment (IEvangelist) [inline docs/azure/index.yml:4]: I love this!!

File: docs/azure/migration/app-service.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ ms.date: 08/11/2018
...
+[App Service](/azure/app-service/overview) is a fully managed compute platform service that's optimized for hosting scalable websites and web applications. This article provides information on how to lift-and-shift an existing application to Azure App Service, modifications to consider, and additional resources for [moving to the cloud](https://azure.microsoft.com/migration/web-applications/). Most ASP.NET websites (Webforms, MVC) and services (Web API, WCF) can move directly to Azure App Service with no changes. Some may need minor changes while others may need some refactoring.
```
Comment (IEvangelist) [inline docs/azure/migration/app-service.md:9]: Are we supposed to be differentiating "ASP.NET" and "ASP.NET Core"

File: docs/azure/migration/app-service.md [hunk 3]
```diff
@@ -31,28 +31,28 @@ Binding | Notes
 --------|--------
 BasicHttp |
 WSHttp |
-WSDualHttpBinding | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled.
-NetHttpBinding | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled for duplex contracts.
-NetHttpsBinding | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled for duplex contracts.
+WSDualHttpBinding | [Web socket support](/azure/app-service/web-sites-configure) must be enabled.
```
Comment (IEvangelist) [inline docs/azure/migration/app-service.md:34]: Consider reformatting this table like so:

```markdown
| Binding | Notes |
|--|--|
| `BasicHttp` |  |
| `WSHttp` |  |
| `WSDualHttpBinding` | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled. | [Web socket support](/azure/app-service/web-sites-configure) must be enabled. |
| `NetHttpBinding` | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled for duplex contracts. | [Web socket support](/azure/app-service/web-sites-configure) must be enabled for duplex contracts. |
| `NetHttpsBinding` | [Web socket support](https://docs.microsoft.com/azure/app-service/web-sites-configure) must be enabled for duplex contracts. | [Web socket support](/azure/app-service/web-sites-configure) must be enabled for duplex contracts. |
| `BasicHttpContextBinding` |  |
| `WebHttpBinding` |  |
| `WSHttpContextBinding` |  |
```

File: docs/azure/migration/app-service.md (hunk 2 above)
Comment (CamSoper) [reply docs/azure/migration/app-service.md:9]: In context, since we mention webforms, I think it should just be "ASP.NET" in this case.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

**Pull request merged**